neo/rawio/plexon2rawio/plexon2rawio.py

```diff
@@ -0,0 +1,498 @@
+            if not achannel_info.m_ChannelEnabled:
+                continue
+
+            # assign to matching stream or create new stream based on signal characteristics
+            rate = achannel_info.m_SamplesPerSecond
+            n_samples = achannel_info.m_NumberOfValues
+            source_id = str(achannel_info.m_Source)
+
+            channel_source = Source(source_id, f'stream@{rate}Hz', rate, n_samples)
+            existing_source = source_characteristics.setdefault(source_id, channel_source)
+
+            # ensure that stream of this channel and existing stream have same properties
+            assert channel_source == existing_source
+
+            ch_name = achannel_info.m_Name.decode()
+            chan_id = f'source{achannel_info.m_Source}.{achannel_info.m_Channel}'
+            dtype = 'int16'
+            units = achannel_info.m_Units.decode()
+            gain = achannel_info.m_CoeffToConvertToUnits
+            offset = 0.  # PL2 files don't contain information on signal offset
+            stream_id = source_id
+            signal_channels.append((ch_name, chan_id, rate, dtype, units, gain, offset, stream_id))
+
+        signal_channels = np.array(signal_channels, dtype=_signal_channel_dtype)
+        self.signal_stream_characteristics = source_characteristics
+
+        # create signal streams from source information
+        signal_streams = []
+        for stream_idx, source in source_characteristics.items():
+            signal_streams.append((source.name, str(source.id)))
+        signal_streams = np.array(signal_streams, dtype=_signal_stream_dtype)
+
+        # pre-loading spike channel_data for later usage
+        self._spike_channel_cache = {}
+        spike_channels = []
+        for c in range(self.pl2reader.pl2_file_info.m_TotalNumberOfSpikeChannels):
+            schannel_info = self.pl2reader.pl2_get_spike_channel_info(c)
+
+            # only consider active channels
+            if not schannel_info.m_ChannelEnabled:
+                continue
+
+            for channel_unit_id in range(schannel_info.m_NumberOfUnits):
+                unit_name = f'{schannel_info.m_Name.decode()}.{channel_unit_id}'
+                unit_id = f'unit{schannel_info.m_Channel}.{channel_unit_id}'
+                wf_units = schannel_info.m_Units
+                wf_gain = schannel_info.m_CoeffToConvertToUnits
+                wf_offset = 0.  # A waveform offset is not provided in PL2 files
+                wf_left_sweep = schannel_info.m_PreThresholdSamples
+                wf_sampling_rate = schannel_info.m_SamplesPerSecond
+                spike_channels.append((unit_name, unit_id, wf_units, wf_gain,
+                                      wf_offset, wf_left_sweep, wf_sampling_rate))
+
+            # pre-loading spiking data
+            schannel_name = schannel_info.m_Name.decode()
+            self._spike_channel_cache[schannel_name] = self.pl2reader.pl2_get_spike_channel_data_by_name(schannel_name)
```

Comment:
I think I would do this on demand and not in init.

Reply:
Good idea, I moved the loading into cache in the respective methods.